pkg/apis/apis/v1alpha1/types.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/third_party/conditions/apis/conditions/v1alpha1"
+)
+
+// ServiceBinding projects APIs and backing implementations into a Workspace.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+type ServiceBinding struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state.
+	// +optional
+	Spec ServiceBindingSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state.
+	// +optional
+	Status ServiceBindingStatus `json:"status,omitempty"`
+}
+
+// ServiceBindingSpec records the APIs and implementations that are to be bound.
+type ServiceBindingSpec struct {
+	// Services describe the APIs and implementations that are to be bound.
+	Services []ServiceSpec `json:"services,omitempty"`
+}
+
+// ServiceSpec describes one API and backing implementation.
+type ServiceSpec struct {
+	Native   *NativeServiceSpec   `json:"native,omitempty"`
```

Comment:
Was not sure if there is a built-in way to enforce that only one of these is set...

Reply:
there is.

Reply:
Am not happy with "native" as a word here. If this is about some catalog, we should call it like that. fromCatalog, global, published

Reply:
Published++